### ISSUE TITLE
Licensing for package CODE vs documentation (text, tutorials)

### ISSUE THREAD

**Issue by lwasser** [pyOpenSci/python-package-guide#39]
from @tupui 

https://github.com/pyOpenSci/python-package-guide/pull/19#discussion_r1065027753

```
About documentation. Some advice to use a different license than code. It's apparently advisable to use a CC one for the doc and something else like MIT/BSD for the code. e.g. in the GNU world they make this distinction: https://www.gnu.org/doc/doc.html
I don't recall any example in the Python sphere though.

Just adding this for completeness (at least what I know). I would not add that to your text which is good I think.

```

we might want to consider clarifying this a bit in our licensing page in the future so users have a better grasp of license issues. that page is it's own page. @sneakers-the-rat might also have some thoughts on expanding this page too.

**Comment by lwasser** [comment]
this is a good issue but we have'nt as a community made a decision on what license to recommend for docs /// we've only decided to use mit for code. i'm going to pull this from the project board until we figure this out so someone can work on it with specific guidance based upon a community decision!

**Comment by lwasser** [comment]
Based on some conversations in Slack, let's add a sentence or two that specifies that the MIT license covers both code and documentation. 

> Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), `

We also don't suggest that people have two separate licenses (for code and documentation) because this risk could cause one license to override the other. 

Adding this content will close this issue. this is a beginner friendly issue!

**Comment by sneakers-the-rat** [comment]
Oh I missed that convo. I don't have strong feelings but I would probably add some discussion on it in the guide. As the in-house copyleft crank, the biggest thing I would think is important to talk about is docs licenses when the software is permissively licensed and the docs contain substantial tutorial content that is useful beyond strictly documenting the project at hand (e.g. sklearn's docs, which are *excellent* for documenting sklearn but also excellent as an entrypoint in hands on machine learning and a lit review).

I may be old fashioned in this era of mass scale text laundering, but I think it's important that people are credited for doing the work and taking the care to share knowledge with each other. I think it would be gross if someone spent a lot of time writing something and then Bigcorp Intl. lifted it wholesale and used it in their Very Profitable Product For Profit Educational Platform without visible attribution.

The MIT license's requirements for attribution are slim and vague:

> The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

(Where "the Software" does include docs)

CC-BY is *much* more prescriptive: https://wiki.creativecommons.org/wiki/License_Versions#Detailed_attribution_comparison_chart

So, e.g. with MIT it would be fine if only the name you included in the copyright notice were included *somewhere* in the docs, including in the fine print on some page linked in a footer, effectively invisible. CC-BY's reasonable attribution requirements arguably require more complete attribution including a link to be shown where the text is shown (a hidden link to a separate fine print attribution page is not reasonable in a webpage, since it's trivial to make such attribution visible).

I am not sure what makes dual licensing challenging - I don't know of a computer-readable way of annotating which license applies to which files in a repo, and I am assuming people aren't bundling the whole docs with the package, but it seems like it would be straightforward to say "the MIT license applies to everything except the /docs directory, which is licensed under CC-{whatever}." Even tho MIT says "and associated documentation files," if you explicitly exempt them then they aren't associated - otherwise you could argue that every text that could conceivably be about the software like other blog posts, talks, times where you posted about it, etc. Are also associated materials. Plus you could just take that clause out, it's your license.

I personally would recommend CC-BY-SA to prevent someone from making the work you provided to the commons into a closed source product. I haven't done so myself but that's usually because I use copyleft licenses.

**Comment by MicahGale** [comment]
> I am not sure what makes dual licensing challenging

The issue is that with [dual licensing](https://en.wikipedia.org/wiki/Multi-licensing) the end user gets to `OR` the license; they need not `AND` them. In practice what this means is that the most permissive license is the de-facto license. This is probably not the developer's intent. 

I think in general the guidelines could be like:

1. If your documentation is simple enough the MIT license will likely suffice
2. If your documentation is in its own repository, or rather complicated consider the CC-BY-SA or CC-BY-NC licenses.
     1. ... warning about multi-licensing ...
3. If in doubt ask your lawyer. 


One thing to keep in mind with all of this. These are just licenses. They don't make unpermitted copying impossible; they just make it so you can sue over it. Are you really willing to spend $10-100k to enforce the nuances between a copyleft and permissive license? Probably not, so in general I would say take the nuance with a grain of salt.

Finally I assume none of us are lawyers, and for sure none of the random people reading the PyOpenSci website are our clients. So we should be cautious about giving anything that could be construed as legal advice.

**Comment by sneakers-the-rat** [comment]
> The issue is that with [dual licensing](https://en.wikipedia.org/wiki/Multi-licensing) the end user gets to OR the license

This is true, but what i was suggesting is not a dual license (two licenses covering the same thing) but one license that covers the docs and one license that covers the software. It's a subtle distinction, and one that would require a bit of discussion (so i think it should be an aside so the main recommendation can remain simple), but i have seen this done a number of times. 

> They don't make unpermitted copying impossible; they just make it so you can sue over it.

Definitely true, but they do have practical effect, and taken to the extreme this would imply "don't even bother with licenses at all because people are just going to do whatever" which would be bad advice. i think it's good to recommend to use software licenses that allow what you intend to allow and prohibit what you intend to prohibit, and hopefully you don't need to sue over it.

> Finally I assume none of us are lawyers, and for sure none of the random people reading the PyOpenSci website are our clients. So we should be cautious about giving anything that could be construed as legal advice.

Also true. we should caveat the whole license section with this. recommending a license is intrinsically making a recommendation about what people should do with relation to the law, but we should make it clear that we are not lawyers.

**Comment by tupui** [comment]
Open source licensing to me is more of a social contract. We tend to forget that there are people behind code and if you don't respect people, they might just stop sharing. That is happening all the time and now even more with the rise of LLM scrapping. So yes, we can abuse the system all day long, but there is going to be some impact on trust and people's willingness to engage.

**Comment by MicahGale** [comment]
> This is true, but what i was suggesting is not a typical meaning of dual license (two licenses covering the same thing) but one license that covers the docs and one license that covers the software. It's a subtle distinction, and one that would require a bit of discussion (so i think it should be an aside so the main recommendation can remain simple), but i have seen this done a number of times.

Right I agree when necessary this is the way to do it. However I think this should be the exception not the rule, otherwise we would be effectively writing a new license for PyOpenSci, which is more effort than necessary I think.

**Comment by sneakers-the-rat** [comment]
yeah, what i was suggesting is having some discussion about it in a collapsible admonition box or something - "if you are writing extensive docs that include tutorial content, here is another thing to think about w.r.t. licensing your work"

**Comment by lwasser** [comment]
I initially thought this would be beginner-friendly, but I now believe that we need folks here who better understand licensing to craft this discussion.  

Is anyone open to crafting a bit of discussion around this topic / creating a pr and then we can review as a community?
In the meantime, i will remove beginner-friendly from the labels :) also 👋🏻 @tupui !!

**Comment by malvikasharan** [comment]
I am looking at the project board shared by Leah at SciPy, and mostly picking those that have some overlap with The Turing Way so that I can connect PyOpenSci to the book and vice versa.

Would it be useful to follow a similar approach as The Turing Way where we apply CC-BY 4.0 for documentation and MIT for code - see here: https://github.com/the-turing-way/the-turing-way/blob/main/LICENSE.md?


I have also edited the chapter that I'd like to link here if useful: https://github.com/the-turing-way/the-turing-way/pull/4300 (the current chapter is here: https://book.the-turing-way.org/reproducible-research/licensing).

**Comment by lwasser** [comment]
@all-contributors please add @malvikasharan for idea, review
